Mathlib/RingTheory/NonUnitalSubring/Basic.lean

```diff
@@ -77,8 +77,8 @@ section NonUnitalSubringClass
 
 /-- `NonUnitalSubringClass S R` states that `S` is a type of subsets `s ⊆ R` that
 are both a multiplicative submonoid and an additive subgroup. -/
-class NonUnitalSubringClass (S : Type*) (R : Type u) [NonUnitalNonAssocRing R]
-    [SetLike S R] extends NonUnitalSubsemiringClass S R, NegMemClass S R : Prop where
+class abbrev NonUnitalSubringClass (S : Type*) (R : Type u) [NonUnitalNonAssocRing R]
+    [SetLike S R] : Prop := NonUnitalSubsemiringClass S R, NegMemClass S R
```

Comment:
This looks like a bad idea to me; `class abbrev` creates loops where it both implies and is implied by the nested classes.

Reply:
I know, I was just experimenting given the terrible performance in the earlier commits.